### ISSUE TITLE
Compiler warning CA1001 is not raised when disposable field is initialized in a private method

### ISSUE THREAD

**Issue by EnricoMassone** [dotnet/roslyn-analyzers#4349]
Environment details:

- operating system: **windows 10 professional build 19041.572**
- analyzers nuget package: **Microsoft.CodeAnalysis.FxCopAnalyzers version 3.3.0**
- **.NET core 3.1 console application**
- code editor: **Microsoft Visual Studio Professional version 16.7.6**

Here is the csproj file: 

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.0">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>

</Project>
```

This class raises the CA1001 compiler warning as expected:

```
public class DisposableFieldHolder
{
    private readonly HttpClient _httpClient = new HttpClient();

    public async Task<string> DownloadWebPageContent() 
    {
      var uri = new Uri("https://github.com/dotnet/roslyn-analyzers");
      var content = await _httpClient.GetStringAsync(uri).ConfigureAwait(false);
      return content;
    }
}
```

This class, instead, does not raise the expected CA1001 compiler warning:

```
public class DisposableFieldHolder
{
    private readonly HttpClient _httpClient = InitHttpClient();

    private static HttpClient InitHttpClient()
    {
      var httpClient = new HttpClient();
      httpClient.DefaultRequestHeaders.Add("User-Agent", "test-application");
      return httpClient;
    }

    public async Task<string> DownloadWebPageContent() 
    {
      var uri = new Uri("https://github.com/dotnet/roslyn-analyzers");
      var content = await _httpClient.GetStringAsync(uri).ConfigureAwait(false);
      return content;
    }
}
```

I would expect the same CA1001 compiler warning in both cases, because the only difference is the way the disposable field is initialized (inline versus in a private method). 

![Compiler_Warning_Raised](https://user-images.githubusercontent.com/29275408/96686657-150ed500-137f-11eb-841c-ca9f63c6467d.png)

![Compiler_Warning_Not_Raised](https://user-images.githubusercontent.com/29275408/96686666-19d38900-137f-11eb-9302-5c91a6fb999e.png)
